### ISSUE TITLE
Enabling GitHub Discussions—or improving the official forum?

### ISSUE THREAD

**Issue by AaronBeaudoin** [odoo/odoo#130047]
**This question is relevant here because this is where it will be seen by the eyes of Odoo developers and not high level support representatives with zero development experience. Read on for why that is important.**

---

The process for asking _technical_ questions regarding Odoo in either a community setting or to the core team is a challenge (I will explain why in a moment). As far as I can tell, the expected solution is to either:

- Use the official help forum: https://www.odoo.com/forum/help-1.
- Open a support ticket via our enterprise account: https://www.odoo.com/help.

The challenge is that neither of these solutions are very good for getting helping with **_technical/developer_** questions. Here are a number of reasons why:

1. The official help forum has a bunch of frustrating limitations that make it difficult to get help. After initially creating a post, it first needs to go through manual approval, delaying the process of getting a response. After it is approved, for those like myself with low XP, I can't even respond under an answer to my own question! Therefore, if the answer isn't sufficient, I have no way to request more info; and of course it also seems rude to simply repost the same question with modifications to try and get a better answer. Recently I posted a question and someone left an answer that didn't address my question, but marked it as the accepted answer—and I can't even change it!
2. Also related to low XP, I can't seem to include links in my post, which is maddening because sometimes it is helpful to link to, for example, a page of the official documentation or an issue here on GitHub. These limitations are unreasonably restrictive.
3. By far the most frustrating issue with the official help forum is the lack of proper support for code blocks in the text editor. I pasted XML from VS Code, but the closing tags in the code were broken, presumably because the editor does not properly escape them. The code also seems to be pasted with all line breaks removed, resulting in an unreadable mess. I do have the option to switch the text to monospace, but the intent seems to be for tiny inline code snippets, not larger blocks. There is also no syntax highlighting. Overall, it is clear that the intent of the official help forum is to cater to _non-technical_ questions.
4. Regarding support tickets: The request seems to always be going through an individual with zero programming knowledge. Therefore, it seems ridiculous to even bother asking a technical question there. And if I do so anyways, the formatting options are even worse than on the help forum—I can only write plain text without even bold or italics!

In contrast, GitHub's markdown editor which I'm using right now has comparatively excellent formatting options, and GitHub imposes no ridiculous limitations which stand in the way of getting proper help. I've used GitHub Discussions on many other public repositories and it is probably one of the best experiences for asking questions I've seen anywhere. Enabling it and directing developers there for technical questions would result in a massively better experience, and any Odoo developer probably already has a GitHub account.

So please, _I'm almost begging_, either enable GitHub Discussions and direct people there, or address the horrific experience over at the official help forum for technical questions. One of the biggest needs for Odoo as an open source ERP is a great experience for getting community help, and I don't personally think the existing options address that need.

**Comment by intero-chz** [comment]
Your reasons to the forum are also my reasons why I avoid the forum as much as possible and personally only help on StackOverflow.

**Comment by gdgellatly** [comment]
Many years ago, openerp ran a simple forum with lots of activity from smart people who couldn't give a toss about karma etc etc interacted. Then they changed to gamify it. All those people left because it took away the interaction and made assistance a transaction and people who cared about made up points came in. My reply count on old forums was around 4000, new system, not sure maybe 1 maybe 0. For what you are looking for - see this video and realise we offer lots of opportunity to our members and contributors. https://www.youtube.com/watch?v=jRZpVR-BizM and browse our youtube channel, website or our mailing lists and github discussions or discord channels. Best 50 EUR you will spend.

**Comment by rdeodoo** [comment]
Hello and thank you for your feedback.

On top of not being technical/development friendly, there is clearly a lack of replies in our forum indeed.
Even for us, Odoo developers, it happens that we do a quick google search about an Odoo technical thing and find the exact same question on the Odoo forum (which was posted multiple years ago sometime) without a reply.

Maybe the reason is a possible bad usability of that forum, as you explained. I will send your feedback to our product owner, they will see if something can be done or take your remarks into consideration for next refactoring. But it's something we already knew.

AFAIC, I believe the technical **questions** and **community** should take place on stackoverflow for simple reasons:
1. The UX is bulletproof, it has been battle tested for longer than any other platform now
2. It's "technical" oriented
3. People have an incentive to be active on it and to reply to questions: they want and will gain points on the n°1 technical worldwide forum. For multiple reasons, people want to contribute on stackoverflow: build a nice profile, be famous, be known, just have the great feeling of being upvoted or selected as the good answer etc

I don't think any other platform would be viable on the long run (I will explain why comparing to previous three points):
- Odoo.com forum:
  1. We could surely have a better UX than what we have now, but for the text editor where you type your question, it would be difficult to have our text editor (wysiwyg) behaves ideally for such technical questions with embed code and copy paste: it's the opposite of what is trying to be achieved as our wysiwyg is mainly oriented for our website builder, which targets non technical users. That would require another text editor, probably a simple text edition with just markdown. I don't think we have that and I don't think we want to focus on something like that at the moment. There is still many things to improve in our wysiwyg before considering such an approach.
  But I am sure we could come up with a solution quite quickly if it was judged necessary/priority.
  2. This could be solved by creating another "technical only" forum
  3. Tech people will never have as much will / incentive to  participate in our forum, that would be too niche (only related to odoo and only an odoo.com account), they would rather work on their stackoverflow profile. Of course, some people will do it, but not as enough as we would need to have multiple reply to a single question IMO.
  And about Odoo developers themselves, it's a thought question, it would be exactly the same as github issue, you will find my opinion about it there: https://github.com/odoo/odoo/pull/102704#issuecomment-1290800819
- Github discussions:
  1. That would solves this point, even if I don't use it I am sure their UX is great
  2. That would solves this point too.
  3. This would probably be less of an issue as people would work on their Github account profile and not just an "odoo.com profile": larger scope, multiple projects, you can reply to questions from multiple languages/repository/technology.
  But are people really active on Github discussions? I mean people other than the ones related to the repository, are they trying to be active or is it basically just people from the project trying to keep up with the questions and replying to it (which we wouldn't do correctly, see point 3. of Stackoverflow above :point_up: )
  
To have a working and active technical forum, on top of a good UX, you need an incentive for people to come and spend their free time replying to people issue, to freely share their experience. People do not share their time and experience without a strong incentive. Everyone has it's own sensitivity and reason to do it but I'd say it's mainly for "e-reputation", where stackoverflow obviously is the winner. Some people even just work on their stackoverflow account just to showcase it on their CV.

But for some reason, Odoo related questions are not active and / or have really poor replies on stackoverflow. It's a bit surprising.
Maybe Github discussion would be a good fit all in all.

**tl;dr**: there is nearly no incentive at all to reply on the Odoo forum, it's by nature, not something we can really change. Only incentive will bring tech people on a forum and motivate them to spend time and share experience for free.
Stackoverflow doesn't seems receptive to Odoo topic. Maybe Github discussions would be a good fit all in all.

This is just my opinion and shouldn't be seen as a general Odoo opinion.

**Comment by gdgellatly** [comment]
@rdeodoo Interesting take. Many years ago we tried to get an OpenERP channel on StackOverflow, we got close but ultimately failed. Certainly I will prefer a peer reviewed SO answer over the cruft in Odoo forums, and there is still a surprising amount of good stuff there.

While Stack Overflow is suffering right now, Odoo IMO is still a good candidate there, partly because it evolves quickly, partly because there is very little online technical info. That means the like of ChatGPT, Copilot etc aren't especially well suited unless you are already quite experienced (or writing migration scripts, copilot rocks at those). Really Copilot firstly stole, then ate stack overflows lunch, it just didn't have much Odoo in it's lunchbox

**Comment by intero-chz** [comment]
The reason why there is so little response to Stack Overflow is, in my opinion, for two reasons. First, some of the questions are extremely poorly written, either because of lacking details or a complete lack of basic knowledge on the part of the questioner. And second, there are currently only 5 to 10 active responders.

Both reasons are partly mutually dependent, because active Stack Overflow users in the area of Odoo questions, could enhance the questions. At the same time, of course, sufficiently detailed questions would be heavenly to generate more activity.

However, from my point of view, there's not much you can do about that, because in the end it's mostly up to the people. Stack Overflow offers all the tools. And that's exactly what the Odoo forum doesn't offer.

Whether Github Discussions will be better accepted, I dare to doubt, for exactly the same reasons @rdeodoo  mentions. Besides, the quality of the Issues is sometimes poor, although the quality has been increasing in recent years.

**Comment by AaronBeaudoin** [comment]
I have two thoughts.

**First**, a brief case in favor of GitHub Discussions. I think it is a better candidate due to the simple fact that it lives directly alongside the Odoo source code repository. Other large projects such as [Next.js](https://github.com/vercel/next.js/discussions) have used it very successfully. However, a lot of projects use GitHub for discussions using only issues. From the README for [Pandas](https://github.com/pandas-dev/pandas#discussion-and-development), for example:

> Most development discussions take place on GitHub in this repo, via the [GitHub issue tracker](https://github.com/pandas-dev/pandas/issues).
> 
> Further, the [pandas-dev mailing list](https://mail.python.org/mailman/listinfo/pandas-dev) can also be used for specialized discussions or design issues, and a [Slack channel](https://pandas.pydata.org/docs/dev/development/community.html?highlight=slack#community-slack) is available for quick development related questions.

The problem with this approach is that your issue tracker is no longer restricted to just issues—it becomes cluttered with a whole bunch of extra questions and discussion that make it more difficult to find the actual issues without relying more and more on clever filtering. Not ideal. Yet it seems obvious to me personally that keeping discussions as close as possible to where the actual code development is happing is a no-brainer. Hence, GitHub Discussions.

Also worth noting is that many other communities (such as Pandas above) have a Slack or Discord server, but Odoo does not—which is another area making it difficult to get help.

**Second**, whether you choose to use StackOverflow or GitHub Discussions or whatever else, it is important that the core Odoo team, documentation, README, and other places actually _direct people_ towards the discussion platform of choice. If we want people to use StackOverflow, then no wonder there are not a lot of people over there if we aren't actually pointing people over there. I think we need to choose a strong platform and then lean into that decision. Update official Odoo material to direct people towards that platform.

**Comment by mikkeschiren** [comment]
Beeing new user of the forum, and the frustration on that I can't post code that is readable, I really would prefer that technical discussions move to Github.
